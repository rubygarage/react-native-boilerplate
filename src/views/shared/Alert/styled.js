import styled from 'styled-components/native';

import Icon from 'views/shared/Icon';

const Wrapper = styled.View`
  position: relative;
  flex-direction: row;
  margin-bottom: 8;
  padding-top: 12;
  padding-right: ${({ onCloseClick }) => (onCloseClick ? 48 : 12)};
  padding-bottom: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const IconWrap = styled.View`
  margin-right: 12;
  flex-shrink: 0;
`;

const ContentArea = styled.View`
  flex-grow: 1;
  padding-right: 36;
  display: flex;
  align-items: flex-start;
`;

const MessageTitle = styled.Text`
  flex-direction: row;
  flex-wrap: wrap;
  margin-top: 2;
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  color: ${({ color }) => color};
`;

const MessageText = styled.Text`
  margin-bottom: 2;
  font-size: 14;
  line-height: 20;
  color: ${({ color }) => color};
`;

const ActionButton = styled.TouchableOpacity`
  flex-direction: row;
  align-items: center;
  max-width: 200;
  height: 36;
  margin-top: 8;
  padding-right: 12;
  padding-left: 12;
  border-radius: 4;
  background-color: ${({ color }) => color};
`;

const ActionIcon = styled(Icon)`
  margin-right: 24;
`;

const ActionButtonText = styled.Text`
  color: ${({ theme }) => theme.colors.white};
  font-size: 14;
  line-height: 20;
  font-weight: bold;
  letter-spacing: -0.2;
`;

const CloseButton = styled.TouchableOpacity`
  position: absolute;
  top: 12;
  right: 12;
  color: ${({ theme }) => theme.colors.white};
`;

export {
  Wrapper,
  IconWrap,
  ContentArea,
  MessageTitle,
  MessageText,
  ActionButton,
  ActionButtonText,
  CloseButton,
  ActionIcon,
};
