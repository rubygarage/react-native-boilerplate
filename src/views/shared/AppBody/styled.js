import styled from 'styled-components/native';

const AppBody = styled.View`
  position: relative;
  flex-grow: 1;
  padding-right: 16;
  padding-left: 16;
  background-color: ${({ theme }) => theme.colors.black};
`;

export default AppBody;
