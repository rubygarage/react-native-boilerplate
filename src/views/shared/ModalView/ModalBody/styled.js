import styled from 'styled-components/native';

const Wrap = styled.View`
  padding-top: 24;
  padding-right: 24;
  padding-left: 24;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
