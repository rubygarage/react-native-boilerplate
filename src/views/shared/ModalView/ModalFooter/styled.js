import styled from 'styled-components/native';

const Wrap = styled.View`
  flex-shrink: 0;
  flex-direction: row;
  justify-content: flex-end;
  padding: 16px;
`;

// eslint-disable-next-line import/prefer-default-export
export { Wrap };
