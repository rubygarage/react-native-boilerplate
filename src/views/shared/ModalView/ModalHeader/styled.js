import styled from 'styled-components/native';

const Wrap = styled.View`
  position: relative;
  flex-direction: row;
  align-items: center;
  flex-shrink: 0;
  padding: 16px 24px;
  border-bottom-color: rgba(82,97,115, 0.12);
  border-bottom-width: 1;
  border-style: solid;
`;

const Title = styled.Text`
  font-size: 16;
  line-height: 24;
  letter-spacing: -0.2;
  font-weight: bold;
  color: ${({ theme }) => theme.colors.black};
`;

const Close = styled.TouchableOpacity`
  flex-shrink: 0;
  margin-right: 16;
`;

export { Wrap, Title, Close };
