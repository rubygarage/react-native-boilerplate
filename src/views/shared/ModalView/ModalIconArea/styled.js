import styled from 'styled-components/native';

import IconView from 'views/shared/Icon';

const Wrap = styled.View`
  justify-content: center;
  align-items: center;
  margin-top: 24;
  margin-bottom: 24;
`;

const Icon = styled(IconView)`
  margin-bottom: 16;
`;

const Text = styled.Text`
  font-size: 14;
  line-height: 20;
  color: ${({ theme }) => theme.colors.black};
`;

export { Wrap, Icon, Text };
