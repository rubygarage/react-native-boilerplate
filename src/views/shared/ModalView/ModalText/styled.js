import styled from 'styled-components/native';

const ModalText = styled.Text`
  margin-bottom: 8;
  font-size: 14;
  line-height: 20;
  color:${({ theme }) => theme.colors.black};
`;

export default ModalText;
