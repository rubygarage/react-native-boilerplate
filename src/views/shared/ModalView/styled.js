import styled from 'styled-components/native';
import ModalView from 'react-native-modal';

const Modal = styled(ModalView)`
  flex-grow: 1;
  justify-content: flex-end;
  margin: 0;
`;

const Overlay = styled.View`
  flex-grow: 1;
  justify-content: flex-end;
`;

const Container = styled.View`
  background-color: ${({ theme }) => theme.colors.white};
  border-top-left-radius: 4;
  border-top-right-radius: 4;
`;

export { Modal, Overlay, Container };
