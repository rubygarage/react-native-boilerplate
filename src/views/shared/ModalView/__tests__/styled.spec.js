import React from 'react';
import renderStyled from 'utils/testHelpers/renderStyled';

import * as S from '../styled';

describe('ModalView styled components', () => {
  describe('S.Overlay matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.Overlay />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('S.Container matches snapshot', () => {
    it('default', () => {
      const wrapper = renderStyled(<S.Container />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
