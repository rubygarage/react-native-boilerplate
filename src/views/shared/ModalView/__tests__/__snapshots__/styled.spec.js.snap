// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalView styled components S.Container matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
      },
    ]
  }
/>
`;

exports[`ModalView styled components S.Overlay matches snapshot default 1`] = `
<View
  style={
    Array [
      Object {
        "flexGrow": 1,
        "justifyContent": "flex-end",
      },
    ]
  }
/>
`;
